src/vision/status_quo/alan_creates_a_hanging_alarm.md

```diff
@@ -0,0 +1,129 @@
+
+[Alan] is a developer on the Bottlerocket project.
+[Bottlerocket] is a Linux-based open-source operating system that is purpose-built by Amazon Web Services for running containers.
+We use our own program [pubsys] to ensure that Bottlerocket update repositories are healthy.
+
+### The TL;DR - What makes Alan Sad
+
+Although the first two of these points are about tokio, they are really about Rust async since tokio serves as the de-facto `std::runtime` for Rust.
+
+- It is confusing and dangerous that multiple tokio runtimes can panic or hang at program runtime.
+- It is challenging that using multiple major versions of tokio (which is allowed by cargo) can fail at runtime.
+- It is unfortunate that we need a 3rd party runtime in order to `block_on` a future, even if we are not trying to write async code.
+
+### Multiple Tokio Runtimes
+
+Bottlerocket uses its own [tough] library to read and write TUF repositories.
+This library was created before async became widespread and [reqwest] changed its main interface to `async`.
+When reqwest switched to async, Alan used the `reqwest::blocking` feature instead of re-writing tough to be an async interface.
+(Maybe Alan [should](https://github.com/awslabs/tough/issues/213) make tough an async interface, but he hasn't yet.)
+In order to provide a non-async interface, `reqwest::blocking` creates a tokio runtime so that it can await futures.
+
+In pubsys Alan created some parallel downloading logic while using the above libraries.
+Without realizing the danger, he created a tokio runtime in pubsys and used futures/await to do this parallelization.
+Surprisingly, in retrospect, this worked... until it didn't.
+
+Recently Alan discovered that his repository verification alarm was hanging.
+The root cause was having multiple tokio runtimes, though Alan don't know what change exposed the issue.
+(Maybe it was a `cargo update`?)
+
+The fix was to eliminate the need for a tokio runtime in the pubsys code path by doing the parallel downloads in a different way
+(first with [threads] for a quick fix, then with a [thread pool]).
+
+[Alan] is surprised and sad since he thought the compiler would help him write safe code.
+Instead the compiler was ignorant of his misuse of the de-facto standard Rust async runtime.
+
+### Addendum: Multiple Tokio Major Versions
+
+Alan is also sad that the cargo package manager doesn't understand the de-facto standard runtime's versioning requirements.
+
+Alan had trouble updating to tokio v1 because:
+- Having two major versions of the tokio runtime can/will cause problems.
+- Cargo does not understand this and allows multiple major versions of tokio.
+
+Ultimately Alan's strategy for this in Bottlerocket is to ensure that only one version of tokio exists in the Cargo.lock.
+This requirement delayed his ability to upgrade to tokio v1 and caused him to use a beta version of actix-web since all depenencies need to agree on tokio v1.
+
+### Addendum 2: Not Easy to Block-On
```

Comment:
Minor point:  is this section an addendum to the above "Multiple Tokio Runtimes" section?  It feels like it's its own section, and so maybe should just be called "Not Easy to Block-On"